pkg/connector/account.go

```diff
@@ -276,13 +286,108 @@ func (o *accountResourceType) Grant(ctx context.Context, principal *v2.Resource,
 		TargetType:       awsSsoAdminTypes.TargetTypeAwsAccount,
 	}
 
-	if _, err := o.ssoAdminClient.CreateAccountAssignment(ctx, inp); err != nil {
+	createOut, err := o.ssoAdminClient.CreateAccountAssignment(ctx, inp)
+	if err != nil {
 		return nil, err
 	}
 
+	attemptCount := 1
+	attemptLogger := l.With(
+		zap.Int("attempt", attemptCount),
+		zap.String("request_id", awsSdk.ToString(createOut.AccountAssignmentCreationStatus.RequestId)),
+		zap.String("principal_id", awsSdk.ToString(createOut.AccountAssignmentCreationStatus.PrincipalId)),
+		zap.String("principal_type", string(createOut.AccountAssignmentCreationStatus.PrincipalType)),
+		zap.String("permission_set_arn", awsSdk.ToString(createOut.AccountAssignmentCreationStatus.PermissionSetArn)),
```

Comment:
I think this is not going to work the way you're expecting. Wouldn't it always report that it's attempt number 1? attemptCount is a privative type so updating the variable later doesn't necessarily updated the value the logger captured. You probably have to have each log take the attempt count as a field.